services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/directory/DiagnosisKeysDirectoryTest.java

```diff
@@ -227,9 +257,17 @@ private void buildDirectoryStructure(Collection<DiagnosisKey> keyseparator, Stri
   }
 
   private Set<String> generateExpectedDirectoryStructure(String country, String date) {
-    Set<String> directoryStructure = IntStream.range(0, 24).mapToObj(Integer::toString)
-        .map(hour -> join(separator, "diagnosis-keys", "country", country, "date", date, "hour", hour, "index"))
-        .collect(Collectors.toSet());
+    Set<String> directoryStructure = new HashSet<>();
+
+    if (country.equals("DE") || country.equals(distributionServiceConfig.getEuPackageName())) {
+      directoryStructure = IntStream.range(0, 24).mapToObj(Integer::toString)
+          .map(hour -> join(separator, "diagnosis-keys", "country", country, "date", date, "hour", hour, "index"))
+          .collect(Collectors.toSet());
+
+    } else {
+      directoryStructure.add(join(separator, "diagnosis-keys", "country", country, "date", date, "hour", "0", "index"));
```

Comment:
If the country is not Germany only directory with hour 0 is produced by the test?

Reply:
Keys from other countries will be distributed without applying the shifting policy. Therefore, they will be distributed right away on submission time, but you are right this could be made more generic.